libs/classifiers/edgetpu/face_mask.py

```diff
@@ -50,35 +34,35 @@ def __init__(self, config):
         self.input_details = self.interpreter.get_input_details()
         self.output_details = self.interpreter.get_output_details()
 
-
-    def inference(self, resized_rgb_image):
+    def inference(self, resized_rgb_image) -> list:
         """
         Inference function sets input tensor to input image and gets the output.
         The interpreter instance provides corresponding class id output which is used for creating result
-
         Args:
-            resized_rgb_image: List of images with shape (no_images, img_height, img_width, channels)
-
+            resized_rgb_image: Array of images with shape (no_images, img_height, img_width, channels)
         Returns:
-            result: List of class id for each input image [0, 0, 1, 1, 0]
-
+            result: List of class id for each input image. ex: [0, 0, 1, 1, 0]
+            scores: The classification confidence for each class. ex: [.99, .75, .80, 1.0]
         """
-        if resized_rgb_image == []:
+        if np.shape(resized_rgb_image)[0] == 0:
             return resized_rgb_image
+        resized_rgb_image = (resized_rgb_image * 255).astype("uint8")
         result = []
-        
-        t_begin = time.perf_counter()
+        net_results = []
         for img in resized_rgb_image:
-            img = img * 255.0
-            img = np.array(img, dtype=np.uint8)
-            input_image = np.expand_dims(img, axis=0)
-            self.interpreter.set_tensor(self.input_details[0]["index"], input_image)
+            img = np.expand_dims(img, axis=0)
+            self.interpreter.set_tensor(self.input_details[0]["index"], img)
+            t_begin = time.perf_counter()
             self.interpreter.invoke()
-            out = self.interpreter.get_tensor(self.output_details[0]["index"])
-            result.append(int(np.argmax(out)))
-            
-        inference_time = time.perf_counter() - t_begin  # Second
-        self.fps = convert_infr_time_to_fps(inference_time)
-        # The function `get_tensor()` returns a copy of the tensor data.
-        # Use `tensor()` in order to get a pointer to the tensor.
-        return result
+            inference_time = time.perf_counter() - t_begin  # Second
+            self.fps = convert_infr_time_to_fps(inference_time)
+            net_output = self.interpreter.get_tensor(self.output_details[0]['index'])[0]
+            net_results.append(net_output)
+            result.append(np.argmax(net_output))  # returns class id
+
+        # TODO: optimized without for
+        scores = []
+        for i, itm in enumerate(net_results):
+            scores.append(1)
```

Comment:
Is it correct that scores is an array of 1s?

Reply:
@alpha-carinae29

Reply:
@alpha-carinae29, I fixed it.